invenio_rdm_records/services/access/service.py

```diff
@@ -749,3 +754,54 @@ def create_guest_access_request(self, identity, token, expand=False, uow=None):
             data=comment,
             uow=uow,
         )
+
+    def _translate_expiration_date(self, value):
+        """Translate secret_link_expiration from ui dropdown value."""
+        if value == 0:
+            return None
+        else:
+            return value
+
+    @unit_of_work()
+    def update_access_settings(
+        self,
+        identity,
+        id_,
+        data,
+        uow=None,
+    ):
+        """Update access settings for a record (resp. its parent)."""
+        record, parent = self.get_parent_and_record_or_draft(id_)
+
+        # Permissions
+        self.require_permission(identity, "manage", record=record)
+
+        expiration_days = self._translate_expiration_date(
+            data["access"]["settings"]["secret_link_expiration"]
+        )
+        data["access"]["settings"]["secret_link_expiration"] = expiration_days
+
+        # Validation
+        data, __ = self.schema_parent.load(
+            data, context=dict(identity=identity), raise_errors=True
+        )
```

Comment:
is it expected for `data` to have the structure of a full parent record (i.e. with all the [other fields](https://github.com/inveniosoftware/invenio-rdm-records/blob/master/invenio_rdm_records/services/schemas/parent/__init__.py#L40-L48) there as well)?
given that a few lines further down it looks like everything except for the `settings` values are ignored anyway, i think this may be a bit overkill.
maybe we should only require the `access` structure?

Reply:
there are some changes done in the other [PR](https://github.com/inveniosoftware/invenio-rdm-records/pull/1385/files#diff-f1ba3f9be59bb3905cbc2e8d700d43e2a22281ff63f9dd08b79b99e07ce26c45R772) @anikachurilova is working on so it might affect how this PR is looking now